tests/rptest/tests/retention_policy_test.py

```diff
@@ -28,6 +28,8 @@ def __init__(self, test_context):
         extra_rp_conf = dict(
             log_compaction_interval_ms=5000,
             log_segment_size=1048576,
+            enable_transactions=True,
```

Comment:
Retention polity test tests cases when we change retention:
 - test_changing_topic_retention
 - test_changing_topic_retention_with_restart
 - test_timequery_after_segments_eviction

While we observed a problem (segments are not being removed) under normal operation so it isn't completely clear if those tests actually test the case we're interested in.

Have you checked that they were also failing before (when idempotency is turned on) and it was caused by the same issue?

Reply:
yes, those test are failing with `acks=1` and without patches from this PR.